src/swh_api_functions.py

```diff
@@ -0,0 +1,23 @@
+#
+# Copyright (c) nexB Inc. and others. All rights reserved.
+# heritedcode is a trademark of nexB Inc.
+# SPDX-License-Identifier: Apache-2.0
+# See http://www.apache.org/licenses/LICENSE-2.0 for the license text.
+# See https://github.com/nexB/heritedcode for support or download.
+# See https://aboutcode.org for more information about nexB OSS projects.
+#
+
+
+import requests
+
+base_swh_api_url = "https://archive.softwareheritage.org/api/1/"
+
+def get_content_information(hash_type, hash):
+
+    """
+    this function returns information like checksums, data_url, file_url, license_url
+    based on hash_type and hash as a json object
+    """
+    api = f"{base_swh_api_url}content/{hash_type}:{hash}/"
+    response = requests.get(api)
+    return response.json()
```

Comment:
Instead of returning a json we should return some type of object which we can obtain by mapping json with some kind of class object, since it returns `checksums`, `data_url`, `file_url`, 'license_url`. We should create a class with at least these as attributes for now.

Reply:
Okay so you we must have class where it'll have attributes which we get in response from SWH api and whenever we get reponse we'll map those keys from json to our class and make new obj fromt that class with these attributes and return that object right??

Reply:
How that class object will benifit our end user when we return it.And how it would be diffrent from an json obj in view of usage?? Even if its a class obj will access out o/p like obj.key right??Will you elaborate a little about its adv and usage plz??

Reply:
Yes

Reply:
You are just returning a json object that you are getting from the API. We can not expect every time data to be there in JSON, it may or may not be there.

Reply:
> You are just returning a json object that you are getting from the API. We can not expect every time data to be there in JSON, it may or may not be there.

I went through their docs and it says as follows:
**API access is over HTTPS.
All API endpoints are rooted at https://archive.softwareheritage.org/api/1/.
Data is sent and received as JSON by default.**

Reply:
> You are just returning a json object that you are getting from the API. We can not expect every time data to be there in JSON, it may or may not be there.

I went through their docs and it says as follows:
**API access is over HTTPS.
All API endpoints are rooted at https://archive.softwareheritage.org/api/1/.
Data is sent and received as JSON by default.**

Reply:
I didn't meant that the data will be in JSON format, I meant that the required data or the attributes you want may or may not be in the JSON

Reply:
Sorry but still not getting you.for now i'll create a class with suppose data_url, checksum etc as attributes and when i'll get response form api i'll make new obj of that class and set those class attributes from the json which i'll get from api in reponse.Ok??

Reply:
Yeah please make a class object.

Reply:
> Yeah please make a class object.

Implemented class and used it in function. Completed updates given by you too.
Please review.
Thanks

Reply:
@pombredanne will you please look into this thread.
Thanks.